EIPS/eip-draft_request_id.md

```diff
@@ -0,0 +1,140 @@
+   * **Then**: `[request_id: P, [hash_0: B_32, hash_1: B_32, ...]]`
+* `NodeData (0x0e)`
+   * **Current:** `[value_0: B, value_1: B, ...]`
+   * **Then**: `[request_id: P, [value_0: B, value_1: B, ...]]`
+* `GetReceipts (0x0f)`
+   * **Current:** `[blockHash_0: B_32, blockHash_1: B_32, ...]`
+   * **Then**: `[request_id: P, [blockHash_0: B_32, blockHash_1: B_32, ...]]`
+* `Receipts (0x10)`
+   * **Current:** `[[receipt_0, receipt_1], ...]`
+   * **Then**: `[request_id: P, [[receipt_0, receipt_1], ...]]`
+
+To elaborate, each command is altered in the following way:
+
+1. Create a list with the `request_id` being the first element.
+2. Make the second element the list that defines the whole command in the current scheme.
+
+This is consistent with the request / response pairs in the `les` protocol.
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+**Q: The efficency gains might encourage clients to flood their peers with too many simultanous requests**
+
+Peers can always throttle or disconnect if they don't feel treated well. This is the same as today.
+
+**Q: If `les` already defines the commands like this, why not just use the `les` protocol?**
+
+In practice, peers that serve the `les` protocol are much harder to find in the network. The reasons for this are varied but might boil down to client defaults, immature implementations or missing incentives.
+
+**Q: Networking works today, isn't this just adding bloat?**
+
+This is adding a single integer per command while at the same time reducing code complexity and improving networking effiency. The addition seems justified.
+
+## Backwards Compatibility
+<!--All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+This EIP extends the `eth` protocol in a backwards incompatible way and requires rolling out a new version, `eth/65`. However, `devp2p` supports running multiple versions of the same wire protocol side-by-side, so rolling out `eth/65` does not require client coordination, since non-updated clients can keep using `eth/64`.
+
+This EIP does not change the consensus engine, thus does *not* require a hard fork.
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Other EIPs can choose to include links to test cases if applicable.-->
+
+TODO
```

Comment:
This could probably be left blank? This EIP does not make any consensus-critical changes

Reply:
(after making a manual edit)

```
>>> from trinity.protocol.eth.commands import GET_NODE_DATA_STRUCTURE
>>> from eth.constants import BLANK_ROOT_HASH
>>> import rlp
>>> rlp.encode([1, [BLANK_ROOT_HASH]], sedes=GET_NODE_DATA_STRUCTURE).hex()
'e301e1a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421'
```

Reply:
:+1:  Yeah, I probably just leave it out unless someone demands it later in the process :sweat_smile: